crates/sui-indexer-alt-framework/src/ingestion/client.rs

```diff
@@ -145,16 +158,24 @@ impl IngestionClient {
                     ),
                 })?;
 
-                self.metrics.total_ingested_bytes.inc_by(bytes.len() as u64);
-                let data: CheckpointData = Blob::from_bytes(&bytes).map_err(|e| {
-                    self.metrics.inc_retry(
-                        checkpoint,
-                        "deserialization",
-                        IngestionError::DeserializationError(checkpoint, e),
-                    )
-                })?;
-
-                Ok(data)
+                Ok(match fetch_data {
+                    FetchData::Raw(bytes) => {
+                        self.metrics.total_ingested_bytes.inc_by(bytes.len() as u64);
+                        Blob::from_bytes(&bytes).map_err(|e| {
+                            self.metrics.inc_retry(
+                                checkpoint,
+                                "deserialization",
+                                IngestionError::DeserializationError(checkpoint, e),
+                            )
+                        })?
+                    }
+                    FetchData::CheckPointData(data) => {
+                        self.metrics
+                            .total_ingested_bytes
+                            .inc_by(size_of_val(&data) as u64);
```

Comment:
`size_of_val` is not going to do the right thing here, because it's not going to look through pointers etc. @bmwill, is there a way to get similar information out from the gRPC client? If not, I would opt for not trying to replicate this metric in the case of RPC clients at all (but then we should update its documentation to mention that).